### PR TITLE
Move files around. Add create endpoint

### PR DESCRIPTION
**Pull request by zhengqixi**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    new_id = len(suppliers) + 1
+    new_name = request_body["name"]
+    if "email" in request_body:
+        email = request_body["email"]
+    else:
+        email = ""
+    if "address" in request_body:
+        address = request_body["address"]
+    else:
+        address = ""
```
Comment (HughHuang96824) [inline app.py:62]: Is it assumed that the request body contains no products info?

File: app.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        new_supplier = Supplier(
+            name=new_name,
+            id=new_id,
+            email=email,
+            address=address
+        )
+        suppliers[new_id] = new_supplier
+        app.logger.info("created new supplier with id {}".format(new_id))
+    except (MissingContactInfo, WrongArgType) as e:
+        return error_response(str(e), 400)
```
Comment (HughHuang96824) [inline app.py:73]: I will refine the exception classes to accommodate this

File: app.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    finally:
+        return new_supplier.to_json(), 200
```
Comment (HughHuang96824) [inline app.py:75]: Returning the entire JSON body does not seem to be necessary. Is it OK to return just an ID with a success msg?

File: app.py (hunk 1 above)
Comment (zhengqixi) [reply app.py:62]: I think we can worry about products later. We'll want to figure out the API for how users would actually send in products.

File: app.py (hunk 3 above)
Comment (zhengqixi) [reply app.py:75]: That's fine. Will change.

File: tests/test_app.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+def test_create_happy_path():
+    response = test_app.put('/supplier',
+                            data=json.dumps({
+                                'name': 'test',
+                                'email': 'nice'
+                            }), content_type='application/json')
+    assert response is not None
+    assert response.status_code == 200
+    assert response.is_json
+    assert 'id' in response.json
```
Comment (HughHuang96824) [inline tests/test_app.py:16]: Could check again if id is None

**Pull request merged**